warehouse/forklift/legacy.py

```diff
@@ -1536,8 +1536,22 @@ def file_upload(request):
                         filename=filename, metadata_filename=metadata_filename
                     ),
                 )
-            with open(temporary_filename + ".metadata", "wb") as fp:
-                fp.write(wheel_metadata_contents)
+            try:
+                with open(temporary_filename + ".metadata", "wb") as fp:
+                    fp.write(wheel_metadata_contents)
```

Comment:
question: do we know what filesystem max length this is supporting? This is only one limit, and we have other limits in play: S3/B2 max length is `<full object path> + filename` must be below 1024 - does our filesystem allow for longer than that?

Reply:
Since this limit is dependent on the filesystem in use (i.e. it _could_ change in the future) I think we still should guard against this error here like this. If we want to additionally guard against the S3/B2 limits, we should do that as well (but note that the filename & path in the linked issue is _much_ shorter than 1024 characters, so it wouldn't have caught this specific instance).